editoast/editoast_authz/src/authorizer.rs

```diff
@@ -17,20 +18,44 @@ pub struct Authorizer<S: StorageDriver> {
 }
 
 impl<S: StorageDriver> Authorizer<S> {
+    /// Initialize an authorizer for the given user.
+    /// If the user doesn't exist, it will be created.
     #[tracing::instrument(skip_all, fields(%user), ret(level = Level::DEBUG), err)]
-    pub async fn try_initialize(
+    pub async fn try_initialize_and_create(
         user: UserInfo,
         regulator: Regulator<S>,
     ) -> Result<Self, Error<S::Error>> {
-        let user_id = regulator
+        match Self::try_initialize(user.identity.clone(), regulator.clone()).await {
+            Ok(authorizer) => Ok(authorizer),
+            Err(Error::UnknownUser(_)) => {
+                regulator
+                    .driver
+                    .ensure_user(&user)
+                    .await
+                    .map_err(Error::Storage)?;
+                Self::try_initialize(user.identity, regulator).await
+            }
+            Err(e) => Err(e),
+        }
```

Comment:
I think this logic could be inlined in the authentication middleware since it's the only place it's used really. (For tests, just using `regulator.ensure_user` does the same.)

Reply:
I don't think I understood your comment. What is the authentication middleware?

Reply:
The middleware in `views/mod.rs` where the headers are parsed